src/ProjectTeamCreator.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import { ProjectTeam } from "./ProjectTeam.sol";
+import "@hats/Interfaces/IHats.sol";
+import "./GnosisSafeProxyFactory.sol";
+import "./GnosisSafeProxy.sol";
+import {Project} from "./tables/Project.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Whitelist} from "./Whitelist.sol";
+import {PaymentSplitter} from "./PaymentSplitter.sol";
+import {HatsModuleFactory} from "@hats-module/HatsModuleFactory.sol";
+import {PassthroughModule} from "./PassthroughModule.sol";
+import {deployModuleInstance} from "@hats-module/utils/DeployFunctions.sol";
+
+contract ProjectTeamCreator is Ownable {
+
+    IHats internal hats;
+
+    ProjectTeam internal projectTeam;
+
+    address internal gnosisSingleton;
+
+    address internal hatsPassthrough;
+
+    GnosisSafeProxyFactory internal gnosisSafeProxyFactory;
+
+    HatsModuleFactory internal hatsModuleFactory;
+
+    Project public table;
+
+    uint256 public projectTeamAdminHatId;
+
+    Whitelist internal whitelist;
+
+    bool public openAccess;
+
+    constructor(address _hats, address _hatsModuleFactory, address _hatsPassthrough, address _projectTeam, address _gnosisSingleton, address _gnosisSafeProxyFactory, address _table, address _whitelist) Ownable(msg.sender) {
+        hats = IHats(_hats);
+        projectTeam = ProjectTeam(_projectTeam);
+        gnosisSingleton = _gnosisSingleton;
+        hatsPassthrough = _hatsPassthrough;
+        gnosisSafeProxyFactory = GnosisSafeProxyFactory(_gnosisSafeProxyFactory);
+        hatsModuleFactory = HatsModuleFactory(_hatsModuleFactory);
+
+        table = Project(_table);
+        whitelist = Whitelist(_whitelist);
+    }
+
+    function setProjectTeamAdminHatId(uint256 _projectTeamAdminHatId) external onlyOwner() {
+        projectTeamAdminHatId = _projectTeamAdminHatId;
+    }
+
+    function setOpenAccess(bool _openAccess) external onlyOwner() {
+        openAccess = _openAccess;
+    }
+
+    function createProjectTeam(string memory adminHatURI, string memory managerHatURI, string memory memberHatURI, string calldata title, uint256 quarter, uint256 year, uint256 MDP, string calldata proposalIPFS, address lead, address[] memory members) external onlyOwner() payable returns (uint256 tokenId, uint256 childHatId) {
+        require(whitelist.isWhitelisted(lead) || openAccess, "Only whitelisted addresses can create a Project Team");
+
+
+        bytes memory safeCallData = constructSafeCallData(lead);
+        GnosisSafeProxy gnosisSafe = gnosisSafeProxyFactory.createProxy(gnosisSingleton, safeCallData);
+
+        //admin hat
+        uint256 teamAdminHat = hats.createHat(projectTeamAdminHatId, adminHatURI, 1, address(gnosisSafe), address(gnosisSafe), true, "");
```

Comment:
nit: can rename this to project admin/manager/contributor to match the projects system a little closer.